src/main/java/hudson/plugins/dimensionsscm/CallbackInstance.java

```diff
@@ -0,0 +1,31 @@
+package hudson.plugins.dimensionsscm;
+
+
+import com.serena.dmclient.api.DimensionsConnection;
+
+public class CallbackInstance {
+
+    private static DimensionsAPICallback dimensionsAPICallback;
+
+    private CallbackInstance() {}
+
+    public static DimensionsAPICallback getInstance(DimensionsConnection connection) {
+
+        if (dimensionsAPICallback != null)
+            return dimensionsAPICallback;
+
+        if (isNewServerVersion(connection))
+            dimensionsAPICallback = new DimensionsAPICallback14();
+        else
+            dimensionsAPICallback = new DimensionsAPICallback12();
+
+        return dimensionsAPICallback;
+    }
+
+    private static boolean isNewServerVersion(DimensionsConnection connection) {
+        String serverVersionStr = (String) connection.getObjectFactory().getServerVersion(0).get(0);
+        Double serverVersion = Double.parseDouble(serverVersionStr != null && !serverVersionStr.isEmpty() ? serverVersionStr : "0.0");
+        return serverVersion.compareTo(14.0) >= 0;
```

Comment:
There has been at least one released version of the Dimensions CM server that identified itself as version `2009` (after its year of release 😃), so suggest treating any number below 1900 as a version, and any number above 1900 as a year (and allow year versions beyond 2019).
```suggestion
        return (serverVersion < 1900.0 && serverVersion >= 14.0) || serverVersion >= 2019.0;
```